merlin-server/constraints-dsl-compiler/src/libs/constraints-edsl-fluent-api.ts

```diff
@@ -325,6 +347,28 @@ export class Spans {
       spansExpression: this.__astNode
     })
   }
+
+
+  /**
+   * Check a constraint for each instance of an activity type.
+   *
+   * @param activityType activity type to check
+   * @param expression function of an activity instance that returns a Constraint
+   * @constructor
+   */
+  public static ForEachActivity<A extends Gen.ActivityType>(
+      activityType: A,
+      expression: (instance: ActivityInstance<A>) => Spans,
+  ): Spans {
+    let alias = 'activity alias ' + Spans.__numGeneratedAliases;
```

Comment:
Trying to justify to myself whether or not it's okay for `Spans.ForEachActivity` and `Constraint.ForEachActivity` to produce overlapping aliases.

I suppose since they're used for distinct use cases, it's probably fine. It would give me a bit of peace of mind if we prefix the alias, like `'spans activity alias ' + Spans__numGeneratedAliases`, but I won't insist if you think it's unnecessary.

Reply:
Okay, will do